tests/contest/contest/src/tests/kill/kill_test.rs

```diff
@@ -0,0 +1,131 @@
+use anyhow::anyhow;
+use test_framework::{Test, TestGroup, TestResult};
+
+use crate::tests::lifecycle::ContainerLifecycle;
+
+fn kill_with_empty_id_test() -> TestResult {
+    let mut container = ContainerLifecycle::new();
+
+    // kill with empty id
+    container.set_id("");
+    let result = match container.kill() {
+        TestResult::Failed(_) => TestResult::Passed,
+        TestResult::Passed => TestResult::Failed(anyhow!("Expected failure but got success")),
+        _ => TestResult::Failed(anyhow!("Unexpected test result")),
+    };
+    container.delete();
+    result
+}
+
+fn kill_non_existed_container() -> TestResult {
+    let mut container = ContainerLifecycle::new();
+
+    // kill for non existed container
+    container.set_id("non-existent-container-id");
+    let result = match container.kill() {
+        TestResult::Failed(_) => TestResult::Passed,
+        TestResult::Passed => TestResult::Failed(anyhow!("Expected failure but got success")),
+        _ => TestResult::Failed(anyhow!("Unexpected test result")),
+    };
+    container.delete();
+    result
+}
+fn kill_created_container_test() -> TestResult {
+    let container = ContainerLifecycle::new();
+
+    // kill created container
+    match container.create() {
+        TestResult::Passed => {}
+        _ => return TestResult::Failed(anyhow!("Failed to create container")),
+    }
+    let result = match container.kill() {
+        TestResult::Passed => TestResult::Passed,
+        TestResult::Failed(_) => {
+            TestResult::Failed(anyhow!("Expected success but got failure"))
+        }
+        _ => TestResult::Failed(anyhow!("Unexpected test result")),
+    };
```

Comment:
I think in both these cases you can use `test_result!` macro to directly return the error without having to do manual match. Same for the cases below.

Reply:
container.kill() return TestResult. type.
So I think we don't have to use test_result macro.
Is it true?